File: backend/models/Enseignant.model.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;
const bcrypt = require('bcrypt');

const enseignantSchema = new Schema({
    nomComplet:{
        type:String,
    },
    cin:{
        type:String,
        required:true
    },
    email:{
        type:String,
    },
    phone:{
        type:String,
        
    },
    photo:{
        type:String
    },
    cv:{
        type:String
    },
    titre:{
        type:String
    }

},{timestamps:true});

enseignantSchema.pre('save',async function(){
    try {
        var enseignant = this;
        const salt = await(bcrypt.genSalt(10));
        const hashpass  =  await bcrypt.hash(enseignant.cin,salt);
        enseignant.cin=hashpass;
       
    } catch (error) {
        throw error;
    }
});

const Enseignant = mongoose.model('Enseignant',enseignantSchema);

module.exports = Enseignant;

File: frontend/lib/models/EnseignantModel.dart

class EnseignantModel{
  String? id;
  String? nomComplet;
  String? cin;
  String? numInsc;
  String? phone;
  String? email;
  String? photo;
  String? titre;
  String? cv;

  EnseignantModel({
    this.id,
    this.nomComplet,
    this.email,
    this.phone,
    this.cin,
    this.cv,
    this.titre,
    this.numInsc,
    this.photo,
  });

  Map<String, dynamic> toJson() => {
    'nomComplet': nomComplet,
    "cin":cin,
    "photo":photo,
    "numInsc":numInsc,
    "phone":phone,
    'email': email,
    "cv":cv,
    "titre":titre
  };

  EnseignantModel.fromJson(Map<String, dynamic> json)
      : nomComplet = json['nomComplet'] as String,
        cin = json['cin'] as String,
        numInsc = json['numInsc'] as String,
        email = json['email'] as String,
        photo = json['photo'] as String,
        cv = json['cv'] as String,
        titre = json['titre'] as String,
        phone = json['phone'] as String;

}